# Study for Child Activity Center Small Business

This dataset explores data that was extracted from CRM app for a small business - activities of the business are as in the header

The dataset first removes all entries before April 2022 and after December 2022. This is to ensure that there is stability in input of information that properly captures the business when it is in its full operational state. As the CSV also initially included bookings made in January 2023 *(prebookings)* those were also removed to try as much as possible to only count bookings that actually had transacted.

After this was done, all entries that occured on Mondays were removed. This is due to the fact that on normal weeks, the business does not operate on Mondays and thus any entries that were recorded on a Monday should be taken as outliers and not counted to the data that gives a proper operational overview of the business. 

There were also entries that were made by the owner/operators of the business that caused dirtying of the data, and those were removed before the work was done to analyse the data that was remaining after all outliers/unclear entries were removed. 

While details of how the data was cleaned can be found in this file, for business insights the reader can scroll down and start reading from **Exploration of Customer Profiles**

In [1]:
import pandas as pd
import numpy as np
data = pd.read_csv('makers.csv')

In [2]:
cleaned_data = data[1:]


In [3]:
cleaned_data["Date"] = pd.to_datetime(cleaned_data["Date"])

/opt/anaconda3/lib/python3.9/site-packages/pandas/core/tools/datetimes.py:1047: UserWarning: Parsing '29-01-2023' in DD/MM/YYYY format. Provide format or specify infer_datetime_format=True for consistent parsing.
  cache_array = _maybe_cache(arg, format, cache, convert_listlike)
/opt/anaconda3/lib/python3.9/site-packages/pandas/core/tools/datetimes.py:1047: UserWarning: Parsing '24-01-2023' in DD/MM/YYYY format. Provide format or specify infer_datetime_format=True for consistent parsing.
  cache_array = _maybe_cache(arg, format, cache, convert_listlike)
/opt/anaconda3/lib/python3.9/site-packages/pandas/core/tools/datetimes.py:1047: UserWarning: Parsing '15-01-2023' in DD/MM/YYYY format. Provide format or specify infer_datetime_format=True for consistent parsing.
  cache_array = _maybe_cache(arg, format, cache, convert_listlike)
/opt/anaconda3/lib/python3.9/site-packages/pandas/core/tools/datetimes.py:1047: UserWarning: Parsing '31-12-2022' in DD/MM/YYYY format. Provide format or specif

In [4]:
day_of_week_array = []
for datetime in cleaned_data["Date"]:
    day = datetime.dayofweek
    day_of_week_array.append(day)
    
day_map = {
    0: "Monday",
    1: "Tuesday",
    2: "Wednesday",
    3: "Thursday",
    4: "Friday",
    5: "Saturday",
    6: "Sunday"
}

cleaned_data["day_week"] = day_of_week_array
cleaned_data['day_week'] = cleaned_data['day_week'].map(day_map)

    

/var/folders/pq/6k_969mj2vz2_1xd9b8z0wr00000gn/T/ipykernel_63595/1486641209.py:16: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  cleaned_data["day_week"] = day_of_week_array
/var/folders/pq/6k_969mj2vz2_1xd9b8z0wr00000gn/T/ipykernel_63595/1486641209.py:17: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  cleaned_data['day_week'] = cleaned_data['day_week'].map(day_map)


In [5]:
cleaned_data = cleaned_data[(cleaned_data['Date'] >= "2022/4/1") & (cleaned_data['Date'] <= "2022/12/31")]

In [6]:
cleaned_data['Date'].unique()

array(['2022-12-31T00:00:00.000000000', '2022-12-24T00:00:00.000000000',
       '2022-12-23T00:00:00.000000000', '2022-12-22T00:00:00.000000000',
       '2022-12-21T00:00:00.000000000', '2022-12-20T00:00:00.000000000',
       '2022-12-18T00:00:00.000000000', '2022-12-17T00:00:00.000000000',
       '2022-12-16T00:00:00.000000000', '2022-12-15T00:00:00.000000000',
       '2022-12-14T00:00:00.000000000', '2022-12-13T00:00:00.000000000',
       '2022-11-12T00:00:00.000000000', '2022-10-12T00:00:00.000000000',
       '2022-09-12T00:00:00.000000000', '2022-08-12T00:00:00.000000000',
       '2022-07-12T00:00:00.000000000', '2022-06-12T00:00:00.000000000',
       '2022-04-12T00:00:00.000000000', '2022-11-30T00:00:00.000000000',
       '2022-11-29T00:00:00.000000000', '2022-11-27T00:00:00.000000000',
       '2022-11-26T00:00:00.000000000', '2022-11-25T00:00:00.000000000',
       '2022-11-24T00:00:00.000000000', '2022-11-23T00:00:00.000000000',
       '2022-11-22T00:00:00.000000000', '2022-11-20

At this point in the data cleaning, all entries that are before April and After December 2022 are removed. Now we need to move forward to remove all data entries that coincide with Monday, as Monday are outlier values that may skew the data (given that the shop is not open on all Mondays

In [7]:
cleaned_data = cleaned_data[cleaned_data['day_week'] != "Monday"]

In [8]:
cleaned_data['day_week'].value_counts()

Sunday       992
Saturday     991
Friday       702
Wednesday    684
Tuesday      589
Thursday     480
Name: day_week, dtype: int64

In [9]:
cleaned_data.shape

(4438, 34)

In [10]:
cleaned_data[(cleaned_data['Client name'] == "Amz") | \
            (cleaned_data["Client name"] == "Aisyah Zainalabidin")].shape

(202, 34)

Above, we have ascertained that Amz and Aisyah have entered a total of 202 entries out of 4438. We need to remove these entries because these entries are unable to reflect the specific customer that they have entered these entries on behalf of, and also because these are done with superadmin rights they are able to book more than a 1 hour slot at a time so these might skew the results

In [11]:
cleaned_data = cleaned_data[(cleaned_data['Client name'] != "Amz") \
                           & (cleaned_data['Client name'] != "Aisyah Zainalabidin")]

In [12]:
cleaned_data.shape

(4236, 34)

Because **4438 - 202 = 4236** we can know ascertain that all entries by **Amz** and **Aisyah Zainaladbidin** have been removed. As the dataset also includes cancelled classes, we should be removing them from the dataset.

In [13]:
cleaned_data[cleaned_data['Is cancelled'] == "Yes"].shape

(833, 34)

In [14]:
cleaned_data = cleaned_data[cleaned_data['Is cancelled'] != "Yes"]
cleaned_data.shape

(3403, 34)

All remaining entries are now entries that are uncancelled and were carried out during the time of study.

# Exploration of Customer Profiles 

In [15]:
cleaned_data['Client name'].value_counts().describe()

count    411.000000
mean       8.250608
std       13.482888
min        1.000000
25%        1.000000
50%        3.000000
75%       10.000000
max      100.000000
Name: Client name, dtype: float64

According to the cleaned dataset, 411 unique customers had sent their children to the facility between April and December 2022. 

From the months of April to December 2022, the average customer send their child to the facility 8.25 hours across 9 Months, translating to slightly less than 1 hour per month. There might be a few factors that are skewing the data 

* Since the business only started proper in April, it could be that the mean would start changing across months. We will look into this later 

* Outliers on the high end might be skewing the data, since the maximum count is 100 hours across the months of April to December 2022. 

We will now move to study 

* Top Customers across April to December
* Trends of unique customer counts across months, to ascertain whether unique customer count is growing

# Exploration of Top Customers


In [16]:
cleaned_data['Client name'].value_counts(normalize = True).sort_values(ascending = False).head(20)

Andrea Chong            0.029490
Kimberly Wong           0.029490
Jeffrey Francisco       0.022117
Melissa Burgess         0.021822
Shirlynn Lukey          0.020348
Scarlet Li              0.020348
Selene Cheng            0.020348
Gabriella Oey           0.017989
Amanda Rabot            0.016514
Laura Tromans           0.015925
Aude Giraud             0.013860
Vivienne Tan            0.013565
Kah nan Khoo            0.012976
Lynn Soh                0.012681
Melissa-Anne Bheem      0.011796
Petrina Soh             0.010911
Jessica Tse             0.010616
Jacintha Plantenberg    0.010321
Celine Trinh            0.010027
Jaime Zhong             0.010027
Name: Client name, dtype: float64

In [17]:
cleaned_data['Client name'].value_counts().sort_values(ascending = False).head(20)

Andrea Chong            100
Kimberly Wong           100
Jeffrey Francisco        75
Melissa Burgess          74
Shirlynn Lukey           69
Scarlet Li               69
Selene Cheng             69
Gabriella Oey            61
Amanda Rabot             56
Laura Tromans            54
Aude Giraud              47
Vivienne Tan             46
Kah nan Khoo             44
Lynn Soh                 43
Melissa-Anne Bheem       40
Petrina Soh              37
Jessica Tse              36
Jacintha Plantenberg     35
Celine Trinh             34
Jaime Zhong              34
Name: Client name, dtype: int64

The top 20 customers from April to December 2022 are shown above. Of note, **Andrea Chong** - who has a certain amount of noteriety, has sent her child to the facility for 100 recorded hours from April to December. She and Kimberly Wong each make up approximately 3% of the billed hours during the time span of this study. 

**Possible Recommendations:**
* Exploration of working with Andrea to be an ambassador for the facility, perhaps doing something in kind on a Monday specifically for a party for her kid (or something along that line) in exchange for social media coverage
* Special thank you notes to the top (x) number of customers to be sent out or referral deals could be worked out with these parents who would definitely be great ambassadors for the facility

In [18]:
before_dec = cleaned_data[cleaned_data['Date'] < '2022/12/1']['Client name'].value_counts()
print(before_dec[before_dec == 1])

Sophia Koh       1
Jaclyn chow      1
Merdin Yeoh      1
Keh Hong Wei     1
Rykiel Lun       1
                ..
Zoey             1
Rhea Ong         1
Charmayne Neo    1
Namirah Naser    1
Ming Xiu Koh     1
Name: Client name, Length: 118, dtype: int64


In [19]:
non_return = before_dec[before_dec == 1].index
december_customers = cleaned_data[cleaned_data['Date'] >= '2022/12/1']['Client name'].value_counts().index

In [20]:
non_return_list = []
for name in non_return:
    if name not in december_customers:
        non_return_list.append(name)
for name in non_return_list:
    print(name)
print("number of customers who did not return:", len(non_return_list))

Sophia Koh
Jaclyn chow
Merdin Yeoh
Keh Hong Wei
Rykiel Lun
Danielle
Sephine Goh
Varun Sajwan
Michael chui
Steph
Sharon Lim
Amanda Björk
Audrey
Tan Wei Ling
Lynn Han
Jialin
Jeanne Lim
Tan wan ting
Charlene Er
Jiaying
Cheryl Heng
Cheryl Lee
Ris Nurul
KM TohKammui
Denise Tan
Pelecia
Pan Ju Khuan
Emilah
Michelle Tan
Huilyung Lee
Jay
Lim Hwee Ying
Apple Ho
Sophia Ng
Joachim Liu
Marouen Zelleg
Audrey Thiam
Phyllis
Valerie Kuek
Evelyn Peng
Timothy
Shane Tan
Peiwen Soh
Carmen Low
Bernice Zhang
Koh Teik Ling
Ann Loh
Christabel Tan
Rina
NG KAI MING MELVIN
Bee leng
Sapna ganglani
Huang Xin
Jerline
Vanathi Eliora Ray
Geraldine John
Tan Weizhao
Xin Jie
Elizabeth Quek
Emma
Sky Kang
Jazreel Lim
Rouba Kouwatli
Yumee
Chloe Dunn-Lee
Wendy How
Stephanie Teo
Sarah Ho
Carine Liew
Connie Lee
Reina
Rachel Wu
Lily Phoa
RFI Iskandar
Neo Xian Zhen
Ankita Varma
Jeffrey Ng
yingyi yeo
Siyun
Idy Chan
Hui Cao
Magdalene Goh
Kush
Huda Hanapiah
FOO WEE YEN
Grace Ng
Joys Tan
Zejun
SYIKIN SHARIFF
Valerie
Eunice Png Jiami

# Non-Returning Customers 
From customers who had sent their children to the facility between April and November 2022, there were 111 customers in total who did not return by December 2022, which means that conversion rate for the facility for first time customers is standing at about 80%

Certain possibilities could be contributing to this 

* Parents might be sending their children for initial trials, and for whatever reason, they might find that it's not for them
* Some of these customers could have come in the early days of the business, where business operations may not have been as smooth yet

**Possible Recommendations**
* Since customer data has email records, it is possible to reach out to these customers to find out why they did not return, and possibly convince them to give the facility another chance. While painful to deal with, knowing that you have customers who are not returning and communicating with them might be a good way to get client feedback as to why for these customers in their minds there is not a product market fit


* Not all feedback needs to be taken into consideration, but it should be heard as it's valuable market intel. For example, some people might find price point to be prohibitive, but if the facility has made a business decision that the economics of its business model requires a certain price point then it should stick by that. That being said, to not ask these customers and find out reasons would be to have the business willingly have a potential blind spot 

In [39]:
email_dictionary = {}


for name in non_return_list:
    if name not in email_dictionary:
        email_dictionary[name] = cleaned_data[cleaned_data['Client name'] == name] \
        ['Client email'].iloc[0]

email_df = pd.Series(email_dictionary)
email_df.head(111)

Sophia Koh              s_khf@hotmail.com
Jaclyn chow             jaccies@gmail.com
Merdin Yeoh          merdinyeoh@gmail.com
Keh Hong Wei       vesperlillet@gmail.com
Rykiel Lun             Lunsuann@gmail.com
                           ...           
Liz Khoo             lizzyissy@hotmail.sg
Zoey                mailforzoey@gmail.com
Rhea Ong              rhea_87@hotmail.com
Namirah Naser    namirahnaser23@gmail.com
Ming Xiu Koh         mingxiukoh@gmail.com
Length: 111, dtype: object

In [41]:
email_df.to_csv('email_list.csv')